meiga/result.py

```diff
@@ -75,43 +86,42 @@ def is_success(self) -> bool:
     def is_failure(self) -> bool:
         return not self._is_success
 
-    def throw(self):
+    def throw(self) -> Union[NoReturn, None]:
         if not self._is_success:
             raise self.value
+        return None
 
-    def unwrap(self):
+    def unwrap(self) -> Union[TS, None]:
         if not self._is_success:
             return None
-        else:
-            return self.value
+        return cast(TS, self.value)
 
-    def unwrap_or(self, failure_value: Any):
+    def unwrap_or(self, failure_value: Any) -> Union[TS, Any]:
         if not self._is_success:
             return failure_value
-        else:
-            return self.value
+        return cast(TS, self.value)
 
-    def unwrap_or_return(self, return_value_on_failure: Any = None):
+    def unwrap_or_return(
+        self, return_value_on_failure: Any = None
+    ) -> Union[TS, NoReturn]:
         if not self._is_success:
             return_value = (
                 self if return_value_on_failure is None else return_value_on_failure
             )
             raise OnFailureException(return_value)
-        else:
-            return self.value
+        return cast(TS, self.value)
 
-    def unwrap_or_throw(self):
-        if self._is_success:
-            return self.value
-        else:
-            self.throw()
+    def unwrap_or_throw(self) -> Union[TS, NoReturn, None]:
```

Comment:
Same as with `get_value`, I am not sure that `None` (or even `NoReturn`!) is valid return type here. The whole semantic is `return value, or fail catastrophically`; I think the return type might simply be `-> TS`

Reply:
Agree